### PR TITLE
Add task-based permissions shortcuts

### PR DESCRIPTION
**Pull request by ajaypratap003**
https://issues.redhat.com/browse/MGDSTRM-6585
https://issues.redhat.com/browse/MGDSTRM-6841

image

image

image

Let's merged app-services-ui PR first
https://github.com/redhat-developer/app-services-ui/pull/326

### CODE REVIEW

**Comment by ajaypratap003** [comment]
> Hi @ajaypratap003 , the Assign permission section in the screenshot above looks good! I'm only able to view the screenshots though, so I don't know what the new "Add permissions" split button looks like when opened, or the info-tip popover. Would you please add a screenshot for those items as well? Thanks!

@lclay2  I've added few more screenshots . Add permission is split button and dropddown will open on click icon.

File: src/modules/Permissions/components/ManagePermissionsDialog/CreatePermissions.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.create-permission__action-list-item {
+  margin-left: 20px;
+  margin-top: 15px;
```
[the remark this answers is not in the document]
Comment (ajaypratap003) [reply src/modules/Permissions/components/ManagePermissionsDialog/CreatePermissions.css:3]: @jgiardino This css used for `add-permission` dropdown alignment and css applied on action-list-item.
image


should I use below pf- variables ?
```
  margin-left: var(--pf-global--spacer--lg); // 1.5rem
  margin-top: var(--pf-global--spacer--md) //1rem;
```

**Review by riccardo-forina — APPROVED** [review]
Let's get this on QA so to visually review it

File: src/modules/Permissions/components/ManagePermissionsDialog/CreatePermissions.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (jgiardino) [reply src/modules/Permissions/components/ManagePermissionsDialog/CreatePermissions.css]: I looked at the current implementation and your examples to try to figure out what approach makes sense. 

I agree with adding `margin-top: var(--pf-global--spacer--md) //1rem;`

Looking at the designs, this component should be left-aligned with the table above it, so there should not be a `margin-left` defined. I suggest removing that line.

**Comment by jgiardino** [comment]
I was in back-to-back meetings today, so didn't get a chance to post this one other comment...

The width of the menu should be wider than the toggle. Right now, they are the same width, and this forces the menu to be much taller than it needs to be due to the wrapping description text. If it's wider, then the height will be reduced. Ideally, the menu would not have vertical scrolling, but without seeing a different width, I'm not sure if that's completely attainable, but it's worth trying to eliminate the vertical scroll bar if possible.

We had a similar use case with this example, where we were able to modify the width of the menu relative to the toggle: https://619e1921b3e38b003afeb2be-cdfgnosvcu.chromatic.com/?path=/story/components-metrics-filterbytopic--does-not-break-with-long-words

image

**Comment by ajaypratap003** [comment]
> I was in back-to-back meetings today, so didn't get a chance to post this one other comment...
> 
> The width of the menu should be wider than the toggle. Right now, they are the same width, and this forces the menu to be much taller than it needs to be due to the wrapping description text. If it's wider, then the height will be reduced. Ideally, the menu would not have vertical scrolling, but without seeing a different width, I'm not sure if that's completely attainable, but it's worth trying to eliminate the vertical scroll bar if possible.
> 
> We had a similar use case with this example, where we were able to modify the width of the menu relative to the toggle: https://619e1921b3e38b003afeb2be-cdfgnosvcu.chromatic.com/?path=/story/components-metrics-filterbytopic--does-not-break-with-long-words
> 
> image

@jgiardino  I've raised a new PR for this scrollbar issue. Please have a look. 
https://github.com/bf2fc6cc711aee1a0c2a/kafka-ui/pull/590

**Pull request merged**